File: src/ogt/draw/tile/chamfers.py
"""openGrid chamfer cutout

Retroengineered from the original STEP file, see ./retroengineer.

"""

import math

import cadquery as cq

from ogt.draw.tile.full import TILE_THICKNESS as FULL_TILE_THICKNESS

INTERSECTION_DISTANCE = 4.2

TILE_CHAMFER = math.sqrt(INTERSECTION_DISTANCE**2 * 2)
TILE_CHAMFER_CUTOUT = (
    cq.Workplane("XY")
    .rect(TILE_CHAMFER, TILE_CHAMFER)
    .extrude(FULL_TILE_THICKNESS)
    .rotate((0, 0, 0), (0, 0, 1), 45)
    .translate((0, 0, 0))
)

File: packages/ogt-web/src/lib/cad/chamfers.ts
import { drawRectangle, type Solid } from "replicad";
import { TILE_THICKNESS } from "./constants";

const INTERSECTION_DISTANCE = 4.2;
const TILE_CHAMFER = Math.sqrt(INTERSECTION_DISTANCE ** 2 * 2);

let cached: Solid | null = null;

export function makeTileChamferCutout(): Solid {
  if (cached) return cached;
  cached = drawRectangle(TILE_CHAMFER, TILE_CHAMFER)
    .rotate(45)
    .sketchOnPlane("XY")
    .extrude(TILE_THICKNESS) as Solid;
  return cached;
}
